x/globalfee/ante/fee.go

```diff
@@ -62,11 +62,11 @@ func (mfd FeeDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, ne
 	// Accept zero fee transactions only if both of the following statements are true:
 	// 	- the tx contains only message types that can bypass the minimum fee,
 	//	see BypassMinFeeMsgTypes;
-	//	- the total gas limit per message does not exceed MaxBypassMinFeeMsgGasUsage,
-	//	i.e., totalGas <= len(msgs) * MaxBypassMinFeeMsgGasUsage
+	//	- the total gas limit per message does not exceed MaxTotalBypassMinFeeMsgGasUsage,
+	//	i.e., totalGas <= MaxTotalBypassMinFeeMsgGasUsage
 	// Otherwise, minimum fees and global fees are checked to prevent spam.
 	containsOnlyBypassMinFeeMsgs := mfd.bypassMinFeeMsgs(msgs)
-	doesNotExceedMaxGasUsage := gas <= uint64(len(msgs))*mfd.MaxBypassMinFeeMsgGasUsage
+	doesNotExceedMaxGasUsage := gas <= mfd.MaxTotalBypassMinFeeMsgGasUsage
 	allowedToBypassMinFee := containsOnlyBypassMinFeeMsgs && doesNotExceedMaxGasUsage
```

Comment:
Hi @alexanderbez , can you please help confirm here, it is ok to refactor `uint64(len(msgs))*mfd.MaxBypassMinFeeMsgGasUsage` to `MaxTotalBypassMinFeeMsgGasUsage ` ?
without upgrade 

`MaxBypassMinFeeMsgGasUsage` is 200_000,
`MaxTotalBypassMinFeeMsgGasUsage` is 1_000_000

Reply:
No, you cannot change gas consumed. If a node is replaying or catching up, it will yield in a consensus fault.

Reply:
ok, if including this refactor to a "hard" upgrade (a major release), there is no extra migration needed in the upgrade handler about this ` MaxTotalBypassMinFeeMsgGasUsage ` refactor?